homeassistant/components/homekit_controller/__init__.py

```diff
@@ -183,6 +183,21 @@ def unique_id(self) -> str:
         return f"homekit-{serial}-aid:{self._aid}"
 
 
+class CharacteristicEntity(HomeKitEntity):
+    """
+    A HomeKit entity that is related to an single characteristic rather than a whole service.
+
+    This is typically used to expose additional sensor, binary_sensor or number entities that don't belong with
+    the service entity.
+    """
+
+    @property
+    def unique_id(self) -> str:
+        """Return the ID of this device."""
+        serial = self.accessory_info.value(CharacteristicsTypes.SERIAL_NUMBER)
+        return f"homekit-{serial}-aid:{self._aid}-sid:{self._iid}-cid:{self._iid}"
```

Comment:
`unique_id`s are per domain/platform so `homekit` probably isn't needed as a prefix.  Since everything else it using it already though it might be best to keep it for consistency.

Reply:
Indeed - did this for consistencies sake.